core/trino-main/src/main/java/io/trino/server/protocol/spooling/SegmentResource.java

```diff
@@ -97,14 +97,13 @@ public Response download(@Context UriInfo uriInfo, @PathParam("identifier") Stri
     public Response acknowledge(@PathParam("identifier") String identifier, @Context HttpHeaders headers)
             throws IOException
     {
-        SpooledSegmentHandle handle = handle(identifier, headers);
         try {
-            spoolingManager.acknowledge(handle);
+            spoolingManager.acknowledge(handle(identifier, headers));
             return Response.ok().build();
         }
         catch (IOException e) {
             return Response.serverError()
-                    .entity(e.toString())
+                    .entity(e.getMessage())
```

Comment:
unrelated to `Generate direct URIs to spooled segments`

Reply:
Can you close your eyes 👀 ?

Reply:
hard